### ISSUE TITLE
Pyramids are clipped when translated

### ISSUE THREAD

**Issue by ahnsws** [napari/napari#1606]
## 🐛 Bug

Hi all, I'm having a absolute blast with napari. Though, I've run into an issue with rendering pyramids with translation:

![napari-multiscale-translate](https://user-images.githubusercontent.com/17461258/92041835-fc068200-ed46-11ea-9373-8f7d56673cde.gif)

## To Reproduce

I'm running the following:
```
OS: Ubuntu 20.04.1 LTS
napari: 0.3.7rc1
Platform: Linux-5.4.0-42-generic-x86_64-with-glibc2.29
Python: 3.8.2 (default, Jul 16 2020, 14:00:26) [GCC 9.3.0]
Qt: 5.15.0
PySide2: 5.15.0
NumPy: 1.19.1
SciPy: 1.5.2
Dask: 2.25.0
VisPy: 0.6.4

GL version: 4.6 (Compatibility Profile) Mesa 20.0.8
MAX_TEXTURE_SIZE: 16384

Plugins:
- napari_plugin_engine: 0.1.7
- svg: 0.1.3
```

To create the two pyramids I run this:

```python
import shutil
import zarr
from skimage import data
from skimage.transform import pyramid_gaussian

chunk_size = 64


def create_zarr(dirpath, image):
    pyramid = pyramid_gaussian(image, downscale=4, max_layer=2, multichannel=True)

    store = zarr.DirectoryStore(dirpath)
    with zarr.group(store) as group:
        series = []
        for i, layer in enumerate(pyramid):
            path = "base" if i == 0 else f"L{i}"
            group.create_dataset(path, data=layer, chunks=(chunk_size, chunk_size, 3))
            series.append({"path": path})

        # zarr metadata
        multiscales = [
            {"version": "0.1", "name": "pyramid", "datasets": series, "type": "pyramid"}
        ]
        group.attrs["multiscales"] = multiscales


try:
    shutil.rmtree("./image0.zarr")
except FileNotFoundError:
    pass

try:
    shutil.rmtree("./image1.zarr")
except FileNotFoundError:
    pass

create_zarr("./image0.zarr", data.astronaut())
create_zarr("./image1.zarr", data.immunohistochemistry())
```

To visualize the two pyramids, I use:

```python
import os

import dask.array as da
import napari


def get_pyramid(dirpath):
    levels = ["base"] + sorted(
        [s for s in os.listdir(dirpath) if not s.startswith(".") and (s != "base")]
    )
    return [da.from_zarr(f"{dirpath}/{level}") for level in levels]


pyramid0 = get_pyramid("./image0.zarr")
pyramid1 = get_pyramid("./image1.zarr")

with napari.gui_qt():
    viewer = napari.Viewer()
    viewer.add_image(
        pyramid0, contrast_limits=[0, 1], multiscale=True,
    )
    viewer.add_image(
        pyramid1, contrast_limits=[0, 1], multiscale=True, translate=(100, 300),
    )
```

## Expected behavior

What I should be seeing is the `immunohistochemistry` pyramid fully within the view area. This seems like it could be related to #1053.

Any help would be greatly appreciated!

**Comment by sofroniewn** [comment]
Possibly related to #1579, we can look in more, thanks for the detailed bug report and glad you're having fun with napari!!

**Comment by sofroniewn** [comment]
Ah @ziyangczi looks like there is already an issue for this - see discussion in #1657

**Comment by liu-ziyang** [comment]
Nice, let me push a quick PR so people can try it out

**Comment by liu-ziyang** [comment]
@wooahn can you please try out this PR and see if it helps? https://github.com/napari/napari/pull/1672

**Comment by sofroniewn** [comment]
This has been fixed by #1673, which includes changes from #1672 and more

**Comment by ahnsws** [comment]
I can confirm that this is fixed in napari 0.4.0. Thanks a bunch!
